app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java

```diff
@@ -327,54 +329,61 @@ public static boolean checkBlukonPacket(byte[] buffer) {
                 } else if (m_minutesBack > 7) {
                     m_minutesBack = 5;
                 }
-                UserError.Log.i(TAG, "read " + m_minutesBack + " mins old trend data");
+                Log.i(TAG, "read " + m_minutesBack + " mins old trend data");
                 for (int i = 0; i < m_minutesBack; i++) {
                     if (--delayedTrendIndex < 0)
                         delayedTrendIndex = 15;
                 }
                 int delayedBlockNumber = blockNumberForNowGlucoseDataDelayed(delayedTrendIndex);
                 currentCommand = "010d0e010" + Integer.toHexString(delayedBlockNumber);//getNowGlucoseData
-                UserError.Log.i(TAG, "getNowGlucoseData backfilling");
+                Log.i(TAG, "getNowGlucoseData backfilling");
             }
             m_getNowGlucoseDataIndexCommand = false;
             m_getNowGlucoseDataCommand = true;
 
         } else if (currentCommand.startsWith("010d0e01") /*getNowGlucoseData*/ && m_getNowGlucoseDataCommand == true && strRecCmd.startsWith("8bde")) {
+        	Log.d(TAG, "Before Saving data: + currentCommand = " + currentCommand);
+            String blockId = currentCommand.substring("010d0e01".length());
+            if(!blockId.isEmpty()) {
+                Log.d(TAG, "Saving data: + blockid = " + Integer.parseInt(blockId, 16));
```

Comment:
I assume there is no way that there can be a number format exception here?

Reply:
I believe that it can not happen, since we are writing this buffer by ourselves. 
But just in case , I have wrapped it in a safer function and tested it again.